Navigation-Examples/Examples/Custom-Waypoints.swift

```diff
@@ -0,0 +1,125 @@
+import Foundation
+import UIKit
+import MapboxCoreNavigation
+import MapboxNavigation
+import MapboxDirections
+import MapboxMaps
+import Turf
+
+class CustomWaypointsViewController: UIViewController, NavigationMapViewDelegate, NavigationViewControllerDelegate {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let origin = CLLocationCoordinate2DMake(37.77440680146262, -122.43539772352648)
+        let firstWaypoint = CLLocationCoordinate2DMake(37.77671493551678, -122.42370544507409)
+        let secondWaypoint = CLLocationCoordinate2DMake(37.76556957793795, -122.42409811526268)
+        let options = NavigationRouteOptions(coordinates: [origin, firstWaypoint, secondWaypoint])
+
+        Directions.shared.calculate(options) { [weak self] (session, result) in
+            switch result {
+            case .failure(let error):
+                print(error.localizedDescription)
+            case .success(let response):
+                guard let route = response.routes?.first, let strongSelf = self else {
+                    return
+                }
+
+                // For demonstration purposes, simulate locations if the Simulate Navigation option is on.
+                // Since first route is retrieved from response `routeIndex` is set to 0.
+                let navigationService = MapboxNavigationService(route: route, routeIndex: 0, routeOptions: options, simulating: simulationIsEnabled ? .always : .onPoorGPS)
+                let navigationOptions = NavigationOptions(navigationService: navigationService)
+                let navigationViewController = NavigationViewController(for: route, routeIndex: 0, routeOptions: options, navigationOptions: navigationOptions)
+                navigationViewController.navigationMapView?.delegate = self
+                navigationViewController.modalPresentationStyle = .fullScreen
+                // Render part of the route that has been traversed with full transparency, to give the illusion of a disappearing route.
+                navigationViewController.routeLineTracksTraversal = true
+
+                strongSelf.present(navigationViewController, animated: true, completion: nil)
+            }
+        }
+    }
+
+    // MARK: - Styling methods
+    func customCircleLayer(with identifier: String, sourceIdentifier: String) -> CircleLayer? {
```

Comment:
I think each method which provides custom shape/layer are of non optional type.
```suggestion
    func customCircleLayer(with identifier: String, sourceIdentifier: String) -> CircleLayer {
```